tests/functional/object/mcg/test_s3_vector.py

```
@@ -1310,3 +1331,592 @@ def test_vector_distance_metrics(self, vector_bucket_factory, mcg_obj):
+                keys=[test_vector_key],
+                vectorBucketName=vector_bucket_name,
+                indexName=policy_index_name,
+            )
+        except ClientError as e:
+            error_code = e.response.get("Error", {}).get("Code", "")
+            if error_code == "AccessDeniedException":
+                logger.info("  ✓ GetVectors denied as expected by policy")
+            else:
+                raise
+        else:
+            assert False, "GetVectors succeeded when policy should have denied it"
```

Ruff (0.15.18)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)